simpleclient/src/main/java/io/prometheus/client/Gauge.java

```diff
@@ -181,6 +181,21 @@ public void setToCurrentTime() {
     public Timer startTimer() {
       return new Timer(this);
     }
+
+    /**
+     * Executes runnable code (ie Java 8 Lambda) and observes a duration of how long it took to run.
+     *
+     * @param timeable Code that is being timed
+     */
+    public void time(Runnable timeable){
```

Comment:
I should have noted the comment 😄 

So you would like an `inc(Runnable runnable)` method that increments when it is finished? I'm not sure how it would behave for `dec` or `set` though...

```
// Convenience methods.
  /**
   * Increment the gauge with no labels by 1.
   */
  public void inc() {
    inc(1);
  }
  /**
   * Increment the gauge with no labels by the given amount.
   */
  public void inc(double amt) {
    noLabelsChild.inc(amt);
  }
  /**
   * Increment the gauge with no labels by 1.
   */
  public void dec() {
    dec(1);
  }
  /**
   * Decrement the gauge with no labels by the given amount.
   */
  public void dec(double amt) {
    noLabelsChild.dec(amt);
  }
  /**
   * Set the gauge with no labels to the given value.
   */
  public void set(double val) {
    noLabelsChild.set(val);
  }
```